climpred/bootstrap.py

```diff
@@ -317,80 +381,199 @@ def _bootstrap_by_stacking(init_pm, control):
+        bootstrapped_pers_skill = xr.concat(
+            pers_skill, dim='iteration', **CONCAT_KWARGS
+        )
+    return (
+        bootstrapped_init_skill,
+        bootstrapped_uninit_skill,
+        bootstrapped_pers_skill,
+    )
+
+
+def _get_resample_func(ds):
+    """Decide for resample function based on input `ds`.
 
     Returns:
-        sig_level: bootstrapped significance levels with
-                   dimensions of init_pm and len(sig) if sig is list
+      callable: `_resample_iterations`: if big and chunked `ds`
+                `_resample_iterations_idx`: else (if small and eager `ds`)
     """
-    if not callable(func):
-        raise ValueError(f'Please provide func as a function, found {type(func)}')
-    warn_if_chunking_would_increase_performance(ds)
-    if isinstance(sig, list):
-        psig = [i / 100 for i in sig]
-    else:
-        psig = sig / 100
-
-    bootstraped_ds = _resample_iterations_idx(
-        ds, iterations, dim=resample_dim, replace=False
-    )
-    bootstraped_results = func(bootstraped_ds, *func_args, **func_kwargs)
-    bootstraped_results = rechunk_to_single_chunk_if_more_than_one_chunk_along_dim(
-        bootstraped_results, dim='iteration'
+    resample_func = (
+        _resample_iterations
+        if (
+            dask.is_dask_collection(ds)
+            and len(ds.dims) > 3
+            # > 2MB
+            and ds.nbytes > 2000000
+        )
+        else _resample_iterations_idx
     )
-    sig_level = bootstraped_results.quantile(dim='iteration', q=psig, skipna=False)
-    return sig_level
+    return resample_func
 
 
-def dpp_threshold(control, sig=95, iterations=500, dim='time', **dpp_kwargs):
-    """Calc DPP significance levels from re-sampled dataset.
+def _maybe_auto_chunk(ds, dims):
+    """Auto-chunk on dimension `dims`.
 
-    Reference:
-        * Feng, X., T. DelSole, and P. Houser. “Bootstrap Estimated Seasonal
-          Potential Predictability of Global Temperature and Precipitation.”
-          Geophysical Research Letters 38, no. 7 (2011).
-          https://doi.org/10/ft272w.
+    Args:
+        ds (xr.object): input data.
+        dims (list of str or str): Dimensions to auto-chunk in.
+
+    Returns:
+        xr.object: auto-chunked along `dims`
 
-    See also:
-        * climpred.bootstrap._bootstrap_func
-        * climpred.stats.dpp
     """
-    return _bootstrap_func(
-        dpp, control, dim, sig=sig, iterations=iterations, **dpp_kwargs
-    )
+    if dask.is_dask_collection(ds) and dims is not []:
+        if isinstance(dims, str):
+            dims = [dims]
+        chunks = [d for d in dims if d in ds.dims]
+        chunks = {key: 'auto' for key in chunks}
+        ds = ds.chunk(chunks)
+    return ds
 
 
-def varweighted_mean_period_threshold(control, sig=95, iterations=500, time_dim='time'):
-    """Calc the variance-weighted mean period significance levels from re-sampled
-    dataset.
+def _chunk_before_resample_iterations_idx(
```

Comment:
this part here got a bit messy, because I moved the stats functions to the end of the file